internal/products.go

```diff
@@ -48,10 +48,55 @@ type Product struct {
 	Description  string       `json:"description"`
 }
 
+// This is set by requestProductsFromRegCodeOrSystem() to indicate whether
+// we're connected to a RMT instance. When set to true. container-suseconnect
+// will add a "credentials" parameter to the URLs return from the server.
+var isSMT bool
```

Comment:
Can we avoid that global mutable state by moving the variable into the new type below? Maybe making `RepositoryUrl` something like:

```go
type Repository struct {
	url   url.URL
	isSMT bool
}
```

Reply:
Yeah, I would love to get rid of that. It's ugly. I just haven't found a way around that you. But maybe that's just my lack for go-knowledge. I'll give it another try.

Reply:
Sounds good, feel free to ask if you need any assistance.